pkg/kubelet/dockertools/docker_manager.go

```diff
@@ -667,18 +667,20 @@ func (dm *DockerManager) runContainer(
 	}
 
 	// Set sysctls if requested
-	sysctls, unsafeSysctls, err := api.SysctlsFromPodAnnotations(pod.Annotations)
-	if err != nil {
-		dm.recorder.Eventf(ref, api.EventTypeWarning, events.FailedToCreateContainer, "Failed to create docker container %q of pod %q with error: %v", container.Name, format.Pod(pod), err)
-		return kubecontainer.ContainerID{}, err
-	}
-	if len(sysctls)+len(unsafeSysctls) > 0 {
-		hc.Sysctls = make(map[string]string, len(sysctls)+len(unsafeSysctls))
-		for _, c := range sysctls {
-			hc.Sysctls[c.Name] = c.Value
+	if container.Name == PodInfraContainerName {
```

Comment:
Was thinking about this alternative as well. The namespaces ipc and net are either the host ones (in case they don't pass validation) or they are shared among the containers of the pod (in which case it's enough to set the sysctls for the infra container).

I can of course recheck the sysctls. Are there other examples following that route?